src/services/codefixes/inferFromUsage.ts

```diff
@@ -192,7 +196,17 @@ namespace ts.codefix {
         const typeNode = type && getTypeNodeIfAccessible(type, declaration, program, host);
         if (typeNode) {
             if (isInJSFile(sourceFile) && declaration.kind !== SyntaxKind.PropertySignature) {
-                changes.tryInsertJSDocType(sourceFile, declaration, typeNode);
+                const parent = isVariableDeclaration(declaration) ? tryCast(declaration.parent.parent, isVariableStatement) : declaration;
+                if (!parent) {
+                    return;
+                }
+                const typeTags = createUnderscoreEscapedMap<JSDocTypeTag | JSDocReturnTag>();
+                const typeTag = isGetAccessorDeclaration(declaration) ? createJSDocReturnTag(typeNode, "") : createJSDocTypeTag(typeNode, "");
+                typeTags.set("TAG" as __String, typeTag);
+                insertMergedJSDoc(
+                    typeTags, changes, sourceFile, parent,
+                    t => t.kind === typeTag.kind && "TAG" as __String,

```

Comment:
IMO using  the type `__String | false` is just confusing -- I was beginning to question my knowledge of operator precedence. Would be more clear if `getKey` took `__String | undefined` and this were `t.kind === typeTag.kind ? "TAG" : undefined`.